Com 24 -Added concepts for adult hiv followup form

rubailly opened the pull request.

rubailly committed:
COM-24: Adding concepts for adult hiv followup form

lauravignoli commented on configuration/concepts/adult_hiv_followup.csv:
```diff
@@ -0,0 +1,72 @@
+uuid,Members,Answers,Fully specified name:fr,Fully specified name:en,Short name:fr,Short name:en,Data class,Data type,_version:1,_order:1002
```
@rubailly Please rename the file to adult_hiv_folloup_form.csv to follow existing naming convention

rubailly replied:
Done.

lauravignoli committed:
COM-24 Format file

lauravignoli commented on configuration/concepts/adult_hiv_followup.csv:
```diff
@@ -0,0 +1,72 @@
+cb279713-3373-44be-91bf-ead37122d098,,,Autres Infections Oportunistes diagnostiquées,Other Oportunistic infections diagnosis,Autres Infections Oportunistes diagnostiquées,Other Oportunistic infections diagnosis,Misc,Text,,
+0d9692be-a4ea-48b8-bf44-3c6c8eb7ecf5,,,Date de diagnostic,Date of diagnosis ,Date de diagnostic,Date of diagnosis ,Misc,Date,,
+c7e66918-f4df-4f98-bd8e-b0ed5b12a4b2,dc10758d-06a2-43f1-b916-a111191e454a;e11eaac0-8c68-406e-aa78-34c2faf436d2;2a1c5417-6947-495b-a504-699eb08f224a;462e1a64-6d23-4590-9393-c5fdcba78a42;ecaf7d8e-161d-4e28-9ce8-b3b7dd9772bb;c6658a5c-91c9-4dd9-b9a2-d002a139dc97;20f7c749-9827-4508-a820-c134ebb3e271;175ae517-88f7-4c77-a07d-5bc149fcab3c;673ce6c2-9d44-48b2-8888-ef656cf1ccfd;e7800399-42d3-4dc6-af11-daa5f ...
+41cd339f-27fb-4acb-8841-74c9ea1069f1,74f83479-387d-46f9-8864-6835ea89c488;2f172c49-0200-4454-8be6-b80ec7e46b8a;5bddec2f-8561-4fe7-868b-801eee1ccb43;7d2ac701-7d53-4b66-a98a-1017b828ccb8;fa9d9a5c-82e1-4865-8376-b9bf4e403994;c7e66918-f4df-4f98-bd8e-b0ed5b12a4b2,,FICHE DU PATIENT AVEC VIH - ADULTE SUIVI,PATIENT WITH HIV - ADULT FOLLOWUP                ,FICHE DU PATIENT AVEC VIH - ADULTE SUIVI,PATIENT ...
```
@rubailly Please add this form to All Obs Templates

rubailly replied:
Done.

lauravignoli commented on configuration/concepts/adult_hiv_followup.csv:
```diff
@@ -0,0 +1,72 @@
+673ce6c2-9d44-48b2-8888-ef656cf1ccfd,,,Toxoplasmose cérébrale (Date),Cerebral Toxoplasmosis,Toxoplasmose cérébrale (Date),Cerebral Toxoplasmosis,Misc,Date,,
+e7800399-42d3-4dc6-af11-daa5f5682b2a,,,Cryptococcose neuro-méningée (Date),Cryptococcal neuro-meningeal disease (Date) ,Cryptococcose neuro-méningée (Date),Cryptococcal neuro-meningeal disease (Date) ,Misc,Date,,
+7b82d88a-f566-4934-b2ad-4f7c7bd48108,,,Méningite bactériennes (Date),Bacterial meningitis (Date) ,Méningite bactériennes (Date),Bacterial meningitis (Date) ,Misc,Date,,
+cb279713-3373-44be-91bf-ead37122d098,,,Autres Infections Oportunistes diagnostiquées,Other Oportunistic infections diagnosis,Autres Infections Oportunistes diagnostiquées,Other Oportunistic infections diagnosis,Misc,Text,,
```
@rubailly  Please use a different uuid for Other Oportunistic infections diagnosis. This uuid is being used for concept "Other": cb279713-3373-44be-91bf-ead37122d098,,,Autres,Other,Autres,Other,Misc,N/A,,

In Lisa's file, "Other" was being used, is there a reason why you updated it to "Other Oportunistic infections diagnosis"?

rubailly replied:
I was getting an error: _08-04-2019 13:10:49 [ERROR] CsvParser - An OpenMRS object could not be constructed or saved from the following CSV line: [512ec210-527e-445a-a564-7a3ddf13ba34, null, null, Autres, Other, Autres, Other, Misc, Text, null, null]
org.openmrs.api.DuplicateConceptNameException: 'Autres' is a duplicate name in locale 'fr'_

and when I changed the uuid to the one that is in the database, it was throwing en error in the UI when the form is opened. The existing one is of Class -Misc and  datatype -N/A
It can't be displayed in the UI. That's why I added a new one.

lauravignoli replied:
This is because Lisa added the wrong uuid for concept "Other". What should be done is to use the concept name "Other" and the right concept will be picked up. Although I think even better would be to use the concept "If other specify". I will chat to Lisa about this.

rubailly committed:
COM-24: Renamed adult_hiv_followup_form file

lauravignoli commented on configuration/concepts/adult_hiv_followup.csv:
```diff
@@ -0,0 +1,72 @@
+57cc5b02-93ee-4101-bf64-af2a3527fe24,,,Bon,Good,Bon,Good,Misc,N/A,,
+482bdb99-9ff7-44f4-b52a-44a16d1e76b1,,,Altéré,Altered,Altéré,Altered,Misc,N/A,,
+b2f650c5-d4be-4ee3-81db-33b7a514306f,,57cc5b02-93ee-4101-bf64-af2a3527fe24;482bdb99-9ff7-44f4-b52a-44a16d1e76b1,Etat,State,Etat,State,Misc,Coded,,
+279583bf-70d4-40b5-82e9-6cb29fbe00b4,,a2065636-5326-40f5-aed6-0cc2cca81ccc;b497171e-0410-4d8d-bbd4-7e1a8f8b504e,Grossesse,Pregnancy,Grossesse,Pregnancy,Misc,Coded,,
```
@rubailly For this concept, pregnancy, I created a new concept on the adult initial form, because the original pregnancy concept was not Coded, and had no answers. What do you think we should do here?

rubailly replied:
I think it makes sense to have two different concepts. The one you created (...which I'm also using in this adult hiv followup form) is a question (...as in "is the patient pregnant?") with a YES and NO answer. The other one should be used for example when the patient is asked the reason of stopping ARVs. One of the reason can be "Pregnancy". This would be a concept of class "Misc" and Datatype "N/A".

rubailly committed:
COM-24: Adding adult_hiv_followup_form to all observation template

lauravignoli commented on configuration/concepts/adult_hiv_followup.csv:
```diff
@@ -0,0 +1,72 @@
+uuid,Members,Answers,Fully specified name:fr,Fully specified name:en,Short name:fr,Short name:en,Data class,Data type,_version:1,_order:1002
```
@rubailly Since concept names can be used instead of uuids I asked Lisa to use concept names so that it was easier to identify the members, answers used. Is there any reason why you used uuids everywhere?

rubailly replied:
Concept names had typos and the import was throwing errors due to not being able to find concepts (...I attached some on the ticket). I decided to use uuids to make sure concepts are found.

lauravignoli replied:
In future let's try to fix the typos instead. It's a good opportunity to do so :)

rubailly replied:
True

lauravignoli commented on configuration/concepts/adult_hiv_followup.csv:
```diff
@@ -0,0 +1,72 @@
+34c25e99-f09b-4514-85b9-e2a9f4962085,,a2065636-5326-40f5-aed6-0cc2cca81ccc;b497171e-0410-4d8d-bbd4-7e1a8f8b504e,Prophylaxie a INH,INH prophylaxis ,Prophylaxie a INH,INH prophylaxis ,Misc,Coded,,
+2f172c49-0200-4454-8be6-b80ec7e46b8a,11a26112-8ceb-4a0c-aa46-46175b57be22;bcf78400-e5a9-4070-90ea-5ff83a9b8bd3;34c25e99-f09b-4514-85b9-e2a9f4962085,,II. Prophylaxie,II. Prophylaxis,II. Prophylaxie,II. Prophylaxis,Misc,N/A,,
+4d1386e7-fdab-4d19-a1c3-b58348b75f4a,,,Protocole anti TB (ex. RHEZ),Pr otocole anti TB (ex. RHEZ) ,Protocole anti TB (ex. RHEZ),Pr otocole anti TB (ex. RHEZ) ,Misc,Text,,
+f13460f6-feb9-4251-9300-68af6ff1bfa1,,,Traitement Candidose orale (ex. Nystatine),treatment Oral candidiasis (eg Nystatin) ,Traitement Candidose orale (ex. Nystatine),treatment Oral candidiasis (eg Nystatin) ,Misc,Text,,
```
@rubailly Please use capital T for the english concepts

rubailly replied:
Done

lauravignoli commented on configuration/concepts/adult_hiv_followup.csv:
```diff
@@ -0,0 +1,72 @@
+c6658a5c-91c9-4dd9-b9a2-d002a139dc97,,,Herpés (Date),Herpes (Date),Herpés (Date),Herpes (Date),Misc,Date,,
+20f7c749-9827-4508-a820-c134ebb3e271,,,Candidose oesophagienne (Date),Esophageal candidiasis (Date),Candidose oesophagienne (Date),Esophageal candidiasis (Date),Misc,Date,,
+175ae517-88f7-4c77-a07d-5bc149fcab3c,,,Isosporose (Date),Isosporosis (Date) ,Isosporose (Date),Isosporosis (Date) ,Misc,Date,,
+673ce6c2-9d44-48b2-8888-ef656cf1ccfd,,,Toxoplasmose cérébrale (Date),Cerebral Toxoplasmosis,Toxoplasmose cérébrale (Date),Cerebral Toxoplasmosis,Misc,Date,,
```
@rubailly Please add (Date) to the english concept names

rubailly replied:
Done

lauravignoli commented on configuration/concepts/child_hiv_initial_form.csv:
```diff
@@ -6,4 +6,4 @@ uuid,Members,Answers,Fully specified name:fr,Fully specified name:en,Short name:
 300fffce-30ef-4c3e-a1be-f6acbe99c243,51d7342d-d8ad-4061-97f3-c32724a47694,,III. ANTECEDENTS DE L'ENFANT,III. CHILD BACKGROUND,III. ANTECEDENTS DE L'ENFANT,III. CHILD BACKGROUND,Misc,N/A,,
 48a724c1-fd24-45da-855c-33fcb4ce9c5d,300fffce-30ef-4c3e-a1be-f6acbe99c243;97b30092-34d5-4be0-9a2b-52739a8ef20a,,FICHE DU PATIENT AVEC VIH - ENFANT,CHILD HIV PATIENT FORM,FICHE DU PATIENT AVEC VIH - ENFANT,CHILD HIV PATIENT FORM,Misc,N/A,,
-9d91cb77-3f10-11e4-adec-0800271c1b75,18351dd1-078e-456c-8c23-915b16e9dfaf;c36a7537-3f10-11e4-adec-0800271c1b75;48a724c1-fd24-45da-855c-33fcb4ce9c5d;1fb2dd86-53b5-4815-9c64-edc081b908d9,,All Observation Templates,All Observation Templates,,,ConvSet,N/A,,
+9d91cb77-3f10-11e4-adec-0800271c1b75,18351dd1-078e-456c-8c23-915b16e9dfaf;c36a7537-3f10-11e4-adec-0800271c1b75;48a724c1-fd24-45da-855c-33fcb4ce9c5d;1fb2dd86-53b5-4815-9c64-edc081b908d9;41cd339f-27fb-4acb-8841-74c9ea1069f1,,All Observation Templates,All Observation Templates,,,ConvSet,N/A,,
```
@rubailly Please add this in the followup form and not here. In a new environment, when this file is run, the followup form would not exist yet, and it would break the import.

The reason why I added the adult initial form here is because the adult initial form is imported before this one, and they were developed at the same time.

rubailly replied:
@lauravignoli -so, in the followup form, on that specific row when I mention the all obs template concept, I'll have to list the uuids for all the form right? Otherwise, it will override the concept and only display this  followup form.

rubailly replied:
@lauravignoli -I made the change [here](https://github.com/jembi/cameroon-bahmni-metadata/pull/28/commits/e5163cfc5452e89ea6fdaeda02e69245d56cdaa1)

rubailly committed:
COM-24: Changit 't' to capital case in 'treatment Oral candidiasis (eg Nystatin)'

rubailly committed:
COM-24: Adding (Date) to the English names

rubailly committed:
COM-24: Moving addition of adult hiv followup form to all obs template into the adult hiv followup form concepts file

lauravignoli committed:
COM-24 Updated other diagnostic concepts on adult hiv followup form

Updated uuid for "Other Oportunistic infections diagnosis" as it was using the concept "Other" uuid, and renamed "Date of diagnosis" to "Other diagnostic date".

lauravignoli commented on configuration/concepts/adult_hiv_followup_form.csv:
```diff
@@ -0,0 +1,71 @@
+uuid,Members,Answers,Fully specified name:fr,Fully specified name:en,Short name:fr,Short name:en,Data class,Data type,_version:1,_order:1002
+259735eb-9844-4ce1-b87d-e78b9e4846fd,,,Poids (kg),Weight (kg),Poids (kg),Weight (kg),Misc,Numeric,,
```
We shouldn't rename the vitals concepts, and we should re-use the existing ones. I will work on this and ask Lisa to create a separate story to translate them.

lauravignoli committed:
COM-24 Updated pregnancy concepts on adult hiv followup form

lauravignoli committed:
COM-24 Fix vitals concepts in adult hiv followup form

Used existing concepts instead of creating new ones.
